confection/__init__.py

```diff
@@ -43,8 +44,12 @@ def before_read(self, parser, section, option, value):
             json_value = srsly.json_loads(value)
             if isinstance(json_value, str) and json_value not in JSON_EXCEPTIONS:
                 value = json_value
-        except Exception:
-            pass
+        except ValueError:
+            if value and value[0] == value[-1] == "'":
+                warnings.warn(
+                    f"The value [{value}] seems to be single-quoted, but values "
+                    "use JSON formatting, which requires double quotes."
+                )
```

Comment:
Just to be sure in case there are other errors in that might occur here.
```suggestion
        except ValueError:
            if value and value[0] == value[-1] == "'":
                warnings.warn(
                    f"The value [{value}] seems to be single-quoted, but values "
                    "use JSON formatting, which requires double quotes."
                )
        except Exception:
            pass
```